exec/java-exec/src/main/java/org/apache/drill/exec/resolver/ResolverTypePrecedence.java

```diff
@@ -78,6 +78,9 @@
     precedenceMap.put(MinorType.INTERVALYEAR, i+= 2);
     precedenceMap.put(MinorType.INTERVAL, i+= 2);
     precedenceMap.put(MinorType.UNION, i += 2);
+    precedenceMap.put(MinorType.MAP, i += 2);
```

Comment:
List and Map cannot have a higher precedence than Map. Also, it is not clear what the precedence of List and Map should be.  Can you post a specific question to the dev list?